### PR TITLE
Add Instructions for Okta Team Mapping Configurations

### PR DESCRIPTION
**Pull request by cneralich**
## Labels

- [ ] inaccuracy
- [ ] clarification
- [x] new docs
- [ ] cosmetic bug - fixing broken text or markup
- [ ] enhancement - changing the website's behavior/layout

### CODE REVIEW

File: content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md
Comment (nfagerlund) [inline content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md:54]: - Markdown headers should have blank lines after them. 
- The order of words in this header is somewhat confusing. How about switching them around a bit? 

```suggestion
## Team Mapping Configuration (Okta)

```

File: content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md
Comment (nfagerlund) [inline content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md:60]: For accessibility reasons, images must include descriptions. Users with low vision rely on these. 

The description goes in the square brackets here, where you currently have the filename repeated. 

When writing a description, focus on what's most important to a low-vision user during this workflow. In this image, it would probably be the fact that the button is next to the text "Attributes (optional)". For the others it'll be something different.

File: content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md
Comment (nfagerlund) [inline content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md:70]: Normal nouns like "user" and "team" should not be capitalized, because they are not brand names or other proper nouns.

File: content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md
Comment (nfagerlund) [inline content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md:28]: The term "team mapping" should not be capitalized. 

I strongly suspect that "Okta group membership" shouldn't be capitalized either, but I don't know for sure because I'm not as familiar with Okta. Is the exact phrase "Okta Group Membership" used within Okta as consistent trade dress? Or no?

When linking _within_ a page, please omit the path and filename of the page -- you only need the part after the `#` character. Also, you've included the `.md` file extension after the `.html` extension, so this link won't even work -- the .md extension is removed during the site build process. (Note that every other link on the site points to a .html file rather than a .html.md file.)

File: content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md
Comment (nfagerlund) [inline content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md:58]: The combination of numbered and bulleted lists here is kind of disorganized. I suggest turning the things that are currently numbered list items into normal paragraphs, and turning the things that are currently un-numbered bullet list items into numbered list items. 

The reason I'm suggesting this is because the "1" and "2" items are just explaining the prerequisites, whereas the two bullet points are the actual steps you perform.

**Review by nfagerlund — CHANGES_REQUESTED** [review]
This has several minor issues that need fixing before it can be merged. 

It also has a bigger issue, which is that you need to add more explanation about _why_ someone would need to do this. 

From what I can tell, it looks like by default my Okta configuration won't get a MemberOf attribute at all (so, manual team management in TFC), and these are the steps necessary to make MemberOf happen. But the text doesn't explain that yet -- it points me at the description of the general behavior, but it doesn't give me the context to understand  which parts of that general behavior will apply, and under which conditions. So, please explain the purpose of these configuration steps as well.

File: content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md
Comment (cneralich) [reply content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md]: I originally did this to keep consistency with the other sections, but I agree that it's not as clear.  I've updated the section name for clarity.

File: content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md
Comment (cneralich) [reply content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md]: Thank you!  I've added more meaningful descriptions to these that should make them more useful.

File: content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md
Comment (cneralich) [reply content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md]: Updated.

File: content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md
Comment (cneralich) [reply content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md]: Thanks.  Updated.

File: content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md
Comment (cneralich) [reply content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md]: Agreed.  Updated.

File: content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md
Comment (dylanegan) [inline content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md:28]:
```suggestion
For information on configuring automated Team Mapping using Okta group membership, please see the [Team Mapping Configuration (Okta)](#team-mapping-configuration-okta-) section below. 
```

I hate that our Markdown generated documentation adds the `-` at the end. 😬

**Review by dylanegan — APPROVED** [review]
This reads well to me the only concern I have is the different look (i.e., red border) in the screenshots from the other parts of the SSO documentation. I'll leave that up to @nfagerlund to decide upon.

File: content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md
Comment (nfagerlund) [inline content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md]:
```suggestion
For information on configuring automated team mapping using Okta group membership, please see the [Team Mapping Configuration (Okta)](#configuration---team-mapping-okta) section below. 
```

File: content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md
Comment (nfagerlund) [inline content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md:28]:
```suggestion
For information on configuring automated team mapping using Okta group membership, please see the [Team Mapping Configuration (Okta)](#team-mapping-configuration-okta-) section below. 
```

File: content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md
Comment (nfagerlund) [inline content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md:61]:
```suggestion

1. Expand the `Attributes` section of the Application configuration (under the `Sign On` tab):
```

File: content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md
Comment (nfagerlund) [inline content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md:62]:
```suggestion
1. Expand the "Attributes" section of the Application configuration (under the "Sign On" tab):
```

File: content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md
Comment (nfagerlund) [inline content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md:66]:
```suggestion
2. Set the "Group Attribute Statements" to the following:
```

Same as previous tweak. The values below can stay the same though, since they're verbatim text or selections that need to be entered  by the user. 👍🏼

File: content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md
Comment (nfagerlund) [inline content/source/docs/cloud/users-teams-organizations/single-sign-on/okta.html.md:74]:
```suggestion
Once these configuration steps have been completed, **all** Okta groups to which a given user belongs will be passed in the SAML assertion upon login to Terraform Cloud, which means that user will get added automatically to any teams within Terraform Cloud for which there’s an **exact** name match.  Importantly, please note that those users will also be removed from any teams that *aren't* included in their assertion. This overrides any manually set team memberships, so whenever a user logs in via SSO, their team membership is adjusted to match their SAML assertion.
```

**Review by nfagerlund — APPROVED** [review]
I've made a few more adjustments, with explanations in the commit messages. With that, this is ready to go! Thank you for the patch and for the follow-ups, especially those image descriptions. 

As for the red border in the SAML assertion screenshot, I don't care enough to block the PR any longer on it. (And the red border in the other shot is 👍🏼, since it's showing the location of a control in a larger context.)

**Pull request merged**
